### PR TITLE
fix: reset docker context after deploy

### LINKED ISSUE #852: DOCKER_CONTEXT incorrectly set with proj.sh
**Describe the bug**

DOCKER_CONTEXT is not set correctly after deploying to the `dev` server.
The command `docker compose --env-file env-local.env`  does not reset the value of `DOCKER_CONTEXT` which is defined (and exported) in the file `env-local.env`.

**Steps to reproduce**

1. Deploy to the `dev` server with `deploy-stack dev`. (Note that `DOCKER_CONTEXT` is `met-dev` after deployment, while `CHOSEN_ENV` is reset to `env-local.env`)
2. Trying to turn down the local stack with `stop-stack`. However, it killed the remote stack despite the ENV file is `env-local.env`

**Expected behavior**

After deploying to the `dev` server, both the `CHOSEN_ENV` and `DOCKER_CONTEXT` should be reset to the local. 

**Screenshots**

**Acceptance criteria**
- [ ] demonstrate the use of correctly setting the Docker context by automatically running a `cleanup` command before the deployment, thus also keeping our VMs clean (so they don't run out of disk space)

**Additional context**

The buggy behavior is found on Linux. Not tested on Mac.

### PR DESCRIPTION
**Pull request by e0**
**Related issue(s) and PR(s)**  
This PR closes #852.

**Type of change**  

- [x] Bug fix (non-breaking change which fixes an issue)
 
**List of changes made**  

- Reset docker context after deploying by: `export DOCKER_CONTEXT=default`

**Screenshot of the fix**  

**Testing**  

- Instructions on how to test
1. Deploy to dev server: `deploy-stack dev`.
2. Verify that the docker context is reset to `default` after deployment: `docker context show`.
3. Verify that commands such as `stop-stack` no longer affects the dev server.

**Further comments**  

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [x] Tests and lint/format validations are passing
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by mihai-sysbio** [comment]
Have you test what happens when `deploy-stack` and then the kills breaks the command with `ctrl+c`?

**Comment by MalinAhlberg** [comment]
Unfortunately, this still causes problems when killing `deploy-stack`, as @mihai-sysbio suspected.
2022-11-01-161839_1312x420_scrot

File: proj.sh [hunk 1]
```diff
@@ -54,6 +54,7 @@ function deploy-stack {
   docker compose --env-file $CHOSEN_ENV -f docker-compose.yml -f docker-compose-remote.yml --project-name metabolicatlas up --detach --build --force-recreate --remove-orphans --renew-anon-volumes
   docker compose --env-file $CHOSEN_ENV -f docker-compose.yml -f docker-compose-remote.yml exec -d pg psql -U postgres -c "alter user postgres with password '${POSTGRES_PASSWORD}';"
   CHOSEN_ENV=$METATLAS_DEFAULT_ENV
+  export DOCKER_CONTEXT=default
```
Comment (MalinAhlberg) [inline proj.sh:57]:
```sh
function deploy-stack {
  trap "unset DOCKER_CONTEXT" RETURN
  CHOSEN_ENV="env-${1:-$LOCALENV}.env"
  generate-data
  docker compose --env-file $CHOSEN_ENV -f docker-compose.yml -f docker-compose-remote.yml --project-name metabolicatlas up --detach --build --force-recreate --remove-orphans --renew-anon-volumes
  docker compose --env-file $CHOSEN_ENV -f docker-compose.yml -f docker-compose-remote.yml exec -d pg psql -U postgres -c "alter user postgres with password '${POSTGRES_PASSWORD}';"
  CHOSEN_ENV=$METATLAS_DEFAULT_ENV
}
```
This works for me, for handling cases when you have killed the deploy with ctrl+c. Would be nice if someone else could test it!

Comment (e0) [reply proj.sh:57]: Thanks @MalinAhlberg! I was thinking about this before but I'm not sure if this captures all types of interruptions, for example `ctrl+d` and `ctrl+z`. Will try to give this a try now.

Comment (e0) [reply proj.sh:58]: I tried this and had to add a variant for `zsh`:

Please give this a try now when you have time. I can verify that it (resetting context) works running `Ctrl + c` but does not work with `Ctrl + \` for me. It is still better than not having any at all though. Thanks for mentioning this @mihai-sysbio and to @MalinAhlberg again for the suggested solution.

Comment (MalinAhlberg) [reply proj.sh:58]: Nice @e0!

For other reviewers: I found a problem with this solution on Ubuntu, am looking into it now.

Comment (MalinAhlberg) [reply proj.sh:58]: This seems to work for both me and @e0  (see [slack thread](https://nbisweden.slack.com/archives/C02NW26HZ/p1667384187045479) for more info).
```
function deploy-stack {
  trap 'unset DOCKER_CONTEXT; trap - INT' INT TERM
  CHOSEN_ENV="env-${1:-$LOCALENV}.env"
  generate-data
  docker compose --env-file $CHOSEN_ENV -f docker-compose.yml -f docker-compose-remote.yml --project-name metabolicatlas up --detach --build --force-recreate --remove-orphans --renew-anon-volumes
  docker compose --env-file $CHOSEN_ENV -f docker-compose.yml -f docker-compose-remote.yml exec -d pg psql -U postgres -c "alter user postgres with password '${POSTGRES_PASSWORD}';"
  CHOSEN_ENV=$METATLAS_DEFAULT_ENV
  export DOCKER_CONTEXT=default
}
```

**Comment by MalinAhlberg** [comment]
@inghylt @nanjiangshu please verify that the latest commit works for you:
```sh
$ source proj.sh
$ deploy-stack dev
ctrl+c
$ docker context ls
```
(`default` should be marked as the current context, not `dev`).
Also please verify that you are able to deploy to dev as usual.

**Review by nanjiangshu — APPROVED** [review]
Works well when I tested on both Linux and MacOS

**Review by MalinAhlberg — APPROVED** [review]
I approve (:

**Pull request merged**